infrastructure/builder/src/build/monoimage.js

```diff
@@ -147,8 +156,54 @@ const generateMonoimageTasks = ({tasks, baseDir, spec, cfg, cmdOptions}) => {
     },
   });
 
-  // TODO: add hooks push for web-ui
-  // TODO: add entry point for docker run for web-ui
+  hooks.push({
+    name: 'web-ui',
+    build: async (requirements, utils, procs) => {
+      const cacheDir = path.join(workDir, 'ui/cache');
+      if (!fs.existsSync(cacheDir)) {
+        fs.mkdirSync(cacheDir);
+      }
+
+      utils.step({title: 'Run Yarn Install'});
+
+      await dockerRun({
+        image: nodeImage,
+        workingDir: '/app/ui',
+        env: ['YARN_CACHE_FOLDER=/ui/cache'],
+        command: ['yarn'],
+        logfile: `${workDir}/yarn.log`,
+        utils,
+        binds: [
+          `${appDir}:/app/ui`,
+          `${cacheDir}:ui/cache`,
+        ],
+        baseDir,
+      });
+
+      utils.step({title: 'Run Yarn Build'});
+
+      await dockerRun({
+        image: nodeImage,
+        workingDir: '/app/ui',
+        command: ['yarn', 'build'],
+        logfile: `${workDir}/yarn.log`,
+        utils,
+        binds: [
+          `${appDir}:/app/ui`,
+        ],
+        baseDir,
+      });
+    },
+    entrypoints: async (requirements, utils, procs) => {
+      // since we ran `yarn build` already, there's no need to run it again
+      // on startup, so remove it from the entrypoint commands
+      Object.keys(procs).forEach(process => {
+        if (process.startsWith('web-ui/')) {
+          procs[process] = procs[process].replace('yarn build && ', '');
+        }
+      });
```

Comment:
This worked for web-server because it has a Procfile - it just had a bit in there I didn't want.

For `web-ui`, we need to set up an entrypoint that will run nginx.  This will work more like it does for the "References" hook.